Frontend/src/components/Form.js

```diff
@@ -134,6 +134,74 @@ const Form = ({ setDisplayCalendar, setTerm, setSchedules, setScheduleCount, set
         });
     };
 
+    const hasDuplicateCourses = (form, term) => {
+        const inputCourses = {};
+        const coursesSubString = new Set([]);
+        // Check for exact matching courses that are duplicates
+        for (const key in form) {
+            if (key.startsWith('course') && form[key].trim() !== '') {
+                const value = form[key].trim();
+                coursesSubString.add(value.split(" ").slice(0, 2).join(" "));
+                if (inputCourses[value]) {
+                    return true;
+                } else {
+                    inputCourses[value] = true;
+                }
+            }
+        }
+        const sectionsPerCourse = amountOfSectionsPerCourse(coursesSubString, term);
+        const allCourses = Object.keys(inputCourses);
+
+        /* Check if number of courses inputed with a shared prefix (department + code)
+           is equal to the amount of sections of that class. 
+           For example: if SYSC 3200 has three sections A, B, and C, the user can input
+           each of the 3 sections and see a schedule, but if the user enters each of the
+           3 sections + SYSC 3200, they will be warned of a duplicate. 
+        */
+        for (let i = 0; i < allCourses.length; i++) {
+            for (let j = i + 1; j < allCourses.length; j++) {
+                const prefix1 = allCourses[i].match(/^([A-Za-z]+ \d+)/)[0];
+                const prefix2 = allCourses[j].match(/^([A-Za-z]+ \d+)/)[0];
+                if (prefix1 === prefix2) {
+                    const count = sectionsPerCourse[prefix1];
+                    if (count && count === countCommonPrefix(allCourses, prefix1)) {
+                        if ((allCourses[i].length > prefix1.length && allCourses[j].length === prefix2.length) ||
+                            (allCourses[j].length > prefix1.length && allCourses[i].length === prefix1.length)) {
+                            return true;
```

Comment:
How can it possible for line 167 to be true, but the function not return true? For example, suppose all options of "SYSC 2006" were selected ("SYSC 2006", "SYSC 2006 A", "SYSC 2006 B", "SYSC 2006 C"). In this case countCommonPrefix will return 4. If this is the case, then the function must return true in one of the iterations of the loop since one of the options  has to have been "SYSC 2006", otherwise the function would have returned true in the previous `for (const key in form)` loop above. 

With that being said, would it not be possible then to simply have:
`for (let i = 0; i < allCourses.length; i++) {
     const prefix = allCourses[i].match(/^([A-Za-z]+ \d+)/)[0];
     const count = sectionsPerCourse[prefix1];
     if (count && count === countCommonPrefix(allCourses, prefix1)) { return true; }
`